tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementItemViewModel.cs

```diff
@@ -2,61 +2,451 @@
+        // TODO: Save to the database before moving the folder.
+        var newLocation = await PickNewLocationForRepositoryAsync();
+
+        if (string.IsNullOrEmpty(newLocation))
+        {
+            _log.Information("The path from the folder picker is either null or empty.  Not updating the clone path");
+            return;
+        }
+
+        if (string.Equals(Path.GetFullPath(newLocation), Path.GetFullPath(ClonePath), StringComparison.OrdinalIgnoreCase))
+        {
+            _log.Information("The selected path is the same as the current path.  Not updating the clone path");
+            return;
+        }
+
+        var newDirectoryInfo = new DirectoryInfo(Path.Join(newLocation, RepositoryName));
+        var currentDirectoryInfo = new DirectoryInfo(Path.GetFullPath(ClonePath));
+
+        try
+        {
+            currentDirectoryInfo.MoveTo(newDirectoryInfo.FullName);
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Cound not move repository to the selected location.");
+            TelemetryFactory.Get<ITelemetry>().Log(
+                EventName,
+                LogLevel.Critical,
+                new RepositoryLineItemEvent(nameof(MoveRepository), RepositoryName));
+        }
+
+        var repository = GetRepositoryReportIfNull(nameof(MoveRepository));
+        if (repository == null)
+        {
+            return;
+        }
+
+        var didUpdate = _dataAccess.UpdateCloneLocation(repository, newDirectoryInfo.FullName);
+
+        if (!didUpdate)
+        {
+            _log.Warning($"Could not update the database.  Check logs");
+        }
+
+        ClonePath = Path.Join(newLocation, RepositoryName);
     }
 
     [RelayCommand]
-    public void DeleteRepository()
+    public async Task DeleteRepository()
     {
-        throw new NotImplementedException();
+        var cantFindRepositoryDialog = new ContentDialog()
+        {
+            XamlRoot = _window.Content.XamlRoot,
+            Title = $"Would you like to delete this repository?",
+            Content = $"Deleting a repository means it will be permanently removed in File Explorer and from your PC.",
+            PrimaryButtonText = "Yes",
+            CloseButtonText = "Cancel",
+        };
+
+        var dialogResult = await cantFindRepositoryDialog.ShowAsync();
+
+        if (dialogResult != ContentDialogResult.Primary)
+        {
+            return;
+        }
+
+        // Remove the repository.
+        // TODO: Check if this location is a repository and the name matches the repo name
+        // in path.
+        if (!string.IsNullOrEmpty(ClonePath)
+            && Directory.Exists(ClonePath))
+        {
+            // Cumbersome, but needed to remove read-only files.
+            foreach (var myFile in Directory.EnumerateFiles(ClonePath, "*", SearchOption.AllDirectories))
+            {
+                File.SetAttributes(myFile, FileAttributes.Normal);
+                File.Delete(myFile);
```

Comment:
we should probably wrap the deletion calls and also the dialog call in a try/catch

Reply:
Done.